src/openvpn/buffer.c

```diff
@@ -442,6 +442,7 @@ format_hex_ex (const uint8_t *data, int size, int maxoutput,
 				    ((size * 2) + (size / (space_break_flags & FHE_SPACE_BREAK_MASK)) * (int) strlen (separator) + 2),
 				    gc);
   int i;
+  ASSERT( maxoutput > 0 || separator != NULL );
```

Comment:
hi,

On Tue, Jun 07, 2016 at 12:52:20PM -0700, Selva Nair wrote:
> > @@ -442,6 +442,7 @@ format_hex_ex (const uint8_t *data, int size, int maxoutput,
> >  				    ((size * 2) + (size / (space_break_flags & FHE_SPACE_BREAK_MASK)) * (int) strlen (separator) + 2),
> >  				    gc);
> >    int i;
> > +  ASSERT( maxoutput > 0 || separator != NULL );
> 
> This will never happen: If maxoutput < 0, alloc_buf will exit with error, if maxoutput == 0 and separator == NULL the strlen() will segfault. 
> If the idea is to avoid possible NULL dereferencing in strlen(), the ASSERT should precede that.

What Selva says.

And for MSVC compatibility, the assignment then needs to be made 
independent from the variable declaration.  Ugh.

gert
-- 
USENET is *not* the non-clickable part of WWW!
                                                           //www.muc.de/~gert/
Gert Doering - Munich, Germany                             gert@greenie.muc.de
fax: +49-89-35655025                        gert@net.informatik.tu-muenchen.de